eth/tracers/firehose.go

```diff
@@ -2794,3 +2831,24 @@ func (m Memory) GetPtr(offset, size int64) []byte {
 	reminder := m[offset:]
 	return append(reminder, make([]byte, int(size)-len(reminder))...)
 }
+
+type blockPrintJob struct {
+	block    *pbeth.Block
+	finality *FinalityStatus
+}
+
+func (f *Firehose) blockPrintWorker() {
+	defer f.flushDone.Done()
+	for job := range f.blockPrintQueue {
+		f.printBlockToFirehose(job.block, job.finality)
+	}
+}
+
+func (f *Firehose) CloseBlockPrintQueue() {
```

Comment:
Document that this is blocking, describe the overall concurrency that is happening here.